src/nspawn/nspawn.c

```diff
@@ -2838,7 +2837,7 @@ static int setup_machine_id(const char *directory) {
          * in the container and our idea of the container UUID will always be in sync (at least if PID 1 in the
          * container behaves nicely). */
 
-        r = id128_read(directory, "/etc/machine-id", ID128_FORMAT_PLAIN, &id);
+        r = id128_get_machine(directory, &arg_uuid);
```

Comment:
This will now return -ENOMEDIUM when the machine ID is null, which triggers `ERRNO_IS_MACHINE_ID_UNSET()` which changes the behavior of this function

Reply:
I think the behavior change should be desired. The change is now split out, and explain the behavior change in the commit message.